app/model/profile.model.js

```diff
@@ -185,6 +189,50 @@ function loadDomainForAssignments(headers, domains) {
     };
 }
 
+// Certificates
+// ============================================================================
+
+function loadCertificatesForUser(headers) {
+    return function(user) {
+        var connectors = userToCertificateResource.getUserToCertificateConnectorsByUserId(user._id, headers);
+        var certificates = certificateResource.getAllCertificates(headers);
+        return Promise.all([connectors, certificates])
+            .then(function() {
+                return matchCertificatesAndConnectors(certificates.value(), connectors.value())
+                    .then(loadCertificateSubEntities(headers));
+            })
+            .then(utils.setFieldForObject(user, 'certificates'));
+    };
+}
+
+function matchCertificatesAndConnectors(certificates, connectors) {
+    return utils.extractPropertiesFromConnectors('certificateId', connectors, ['skills', 'dateFrom', 'dateTo', 'description', 'updatedAt'])
+        .then(utils.matchListAndObjectIds(certificates));
+}
+
+function loadCertificateSubEntities(headers) {
+    return function(certificates) {
+        return new Promise(function(resolve) {
+            var skills = skillResource.getAllSkills(headers);
```

Comment:
If you only have one promise, there is no need to use Promise.all. Do:

skillResource.getAllSkills(headers)
   .then(function(skills) {
     .....
    });